### ISSUE TITLE
Will cause the header to be misaligned

### ISSUE THREAD

**Issue by cipchk** [EnixCoda/Gitako#74]
png1:

![image](https://user-images.githubusercontent.com/2987467/87167866-38e26980-c300-11ea-96ad-0154e0fa65e4.png)

png2:

![image](https://user-images.githubusercontent.com/2987467/87168022-747d3380-c300-11ea-98c9-244a47cf3672.png)

Looks like it was caused here [content.scss#L30](https://github.com/EnixCoda/Gitako/blob/develop/src/content.scss#L30)

**Comment by EnixCoda** [comment]
Hi @cipchk 

I cannot reproduce this. Navbar and header are well-aligned on my side.

![image](https://user-images.githubusercontent.com/7480839/87214375-74138580-c35e-11ea-8ba2-7849424ee932.png)

The header, which is in light background, is expected to be 1280px wide.

Please check if you are using any other GitHub extensions.

**Comment by cipchk** [comment]
This should be caused by [refined-github](https://github.com/sindresorhus/refined-github/issues/3336), closed it.

**Comment by fregante** [comment]
Why is Gitako affecting the alignment of the header? It's not Gitako’s purpose to do that.

This also **mis**aligns the header in some cases, having the opposite effect:


# Regular (no extensions)

![reg](https://user-images.githubusercontent.com/1402241/87233958-917f3880-c3cc-11ea-83ee-47d4e784527f.png)


# Gitako

Notice the misaligned header and commit content

![gitako](https://user-images.githubusercontent.com/1402241/87233960-9512bf80-c3cc-11ea-9036-6632bf72871f.png)

**Comment by fregante** [comment]
You can make space for the sidebar without changing the alignment.

Related code: https://github.com/EnixCoda/Gitako/commit/0a957c97da3a3b53f899af72744f5e98151591e3

https://github.com/EnixCoda/Gitako/blob/639b671b677aed7d787f0c9b392651f1cb8dec08/src/content.scss#L29-L41

**Comment by EnixCoda** [comment]
hi @fregante 

If not shrink the header, Gitako's icon will hover over GitHub's. See discussions in #47 

I admit that it is not what Gitako supposed to do as a file tree extension. But I do not have alternative methods to show Gitako's icon yet. Octotree's method is great but copying that is not a preferred option.

> This also misaligns the header in some cases, having the opposite effect:

But it works very well in most cases, except in diff pages. The behavior in diff page is acceptable to me BTW.

I've seen a lot of people complaining about the official alignment, which also pushed me to align it in this way.

**Comment by fregante** [comment]
Indeed, apparently someone else mentioned the same problem.

Instead of centering the whole header, you could move the logo to the left with just:

```css
.Header-link {
    margin-left: 40px;
}
```

Also I'd suggest to move Gitako’s icon a little lower too so it's aligned with GitHub’s logo

**Comment by EnixCoda** [comment]
I see, thank you for the suggestion!

Repositioning the icon is definitely necessary, I'll try to figure out a good solution and make centering header an optional feature.

**Comment by MCOfficer** [comment]
PSA: if you're looking for a quick patch, add this to "Custom CSS" in your RefinedGithub settings (or create a userstyle):
```css
body:not(.full-width):not(.enterprise) .repohead:not(.gisthead) > div,
body:not(.full-width):not(.enterprise) .repohead:not(.gisthead) > nav
{
	padding-left: 0 !important;
}
```

**Comment by MCOfficer** [comment]
I'm sorry to say c2fb67fb373b73f170ac7ce38b4d6bc27e8ea4b2 did not make it better:

![image](https://user-images.githubusercontent.com/22377202/87913568-99477880-ca6f-11ea-8171-6d5de70a349b.png)

This fixed it for me, but i assume it can cause issues on small screens:
```css
.gitako-ready main > :first-child > * {
	max-width: 100% !important;
}
```

**Comment by EnixCoda** [comment]
@MCOfficer Did you resize the window before this happens? I cannot reproduce it.

**Comment by MCOfficer** [comment]
@EnixCoda it works with every window width larger than ~1400px. Running firefox 78.0.2 with *only* Refined Github and Gitako installed.

**Comment by EnixCoda** [comment]
@MCOfficer Looks good on my side, using FF of the same version, but without Refined GitHub. I believe it is because that Refined GitHub is also injecting styles to change GitHub's layout and it conflicts with Gitako's.

I'll make the style injection as an optional feature in the next release.

**Comment by tranhl** [comment]
> PSA: if you're looking for a quick patch, add this to "Custom CSS" in your RefinedGithub settings (or create a userstyle):
> 
> ```css
> body:not(.full-width):not(.enterprise) .repohead:not(.gisthead) > div,
> body:not(.full-width):not(.enterprise) .repohead:not(.gisthead) > nav
> {
> 	padding-left: 0 !important;
> }
> ```

This didn't work for me, but the premise behind the fix remains valid, which pointed me in the right direction. I inspected the element that was causing the mis-alignment, and applied `padding-left: 0 !important;` to it.

This is the CSS that worked for me:
```css
body:not(.full-width):not(.enterprise) main > .hide-full-screen > *
{
	padding-left: 0 !important;
}
```

Edit: For clarity, I applied these styles in Refined Github's CSS override preferences, since the alignment issue seems to be caused by styling conflicts between these two extensions.

**Comment by MCOfficer** [comment]
> Edit: For clarity, I applied these styles in Refined Github's CSS override preferences, since the alignment issue seems to be caused by styling conflicts between these two extensions.

Yep, it's conflicting styles. [This css](https://github.com/EnixCoda/Gitako/issues/74#issuecomment-660870690) should also work.

**Comment by sondr3** [comment]
It looks like this on Firefox, not sure if it is related but it is really cumbersome. I'm not sure if this is a general problem on Firefox or just me though. 

![Screenshot from 2020-11-01 00-42-32](https://user-images.githubusercontent.com/2280539/97792192-2d7ebb00-1bdb-11eb-8775-9f50041cf8bd.png)

Thanks for the extension, really enjoying it! :)

**Comment by EnixCoda** [comment]
Hi @sondr3 

Thanks for using Gitako! :)

I did not reproduce the issue in the latest version of Firefox for macOS/Windows 10/Ubuntu. Please try disable other addons to see if any of them is conflicted with Gitako's styles.

**Comment by sondr3** [comment]
I figured it out, it was actually both me and Gitako. I somehow managed to drag the sidebar into getting a negative width. By dragging the sidebar too much to the left, `--gitako-width` became `-100px` which caused this issue, I think it has to do with the sidebar I have open in Firefox because it looks like it keeps changing the width even when I'm outside the actual browser window.

**Comment by EnixCoda** [comment]
Ah, interesting! I reproduced it successfully. Didn't know mouse event could fire even have moved out of the viewport. I played with it for a while and found it also happen in Chrome.

I'll set a threshold to prevent it in the next patch release.

Thank you very much!

**Comment by EnixCoda** [comment]
Fixed in `v2.4.1`
